File: app/src/main/java/fr/xgouchet/chronorg/data/formatters/DurationFormatter.java
package fr.xgouchet.chronorg.data.formatters;

import android.support.annotation.NonNull;

import org.joda.time.Duration;
import org.joda.time.format.PeriodFormatter;
import org.joda.time.format.PeriodFormatterBuilder;

/**
 * @author Xavier Gouchet
 */
public class DurationFormatter implements Formatter<Duration> {

    PeriodFormatter pf = new PeriodFormatterBuilder()
            .appendYears().appendSuffix("y").appendSeparatorIfFieldsBefore(", ")
            .appendMonths().appendSuffix("m").appendSeparatorIfFieldsBefore(", ")
            .appendDays().appendSuffix("d").appendSeparatorIfFieldsBefore(" / ")
            .appendHours().appendSuffix(":")
            .appendMinutes().appendSuffix("'")
            .appendSeconds().appendSuffix("\"")
            .toFormatter();

    @Override public String format(@NonNull Duration duration) {
        return pf.print(duration.toPeriod());
    }
}

File: khronorg/src/main/java/fr/xgouchet/khronorg/commons/formatters/DurationFormatter.kt
package fr.xgouchet.khronorg.commons.formatters

import org.joda.time.ReadableDuration
import org.joda.time.format.PeriodFormatterBuilder

/**
 * @author Xavier F. Gouchet
 */
object DefaultDurationFormatter : Formatter<ReadableDuration> {


    internal val pf = PeriodFormatterBuilder()
            .appendYears().appendSuffix("y").appendSeparatorIfFieldsBefore(", ")
            .appendMonths().appendSuffix("m").appendSeparatorIfFieldsBefore(", ")
            .appendDays().appendSuffix("d").appendSeparatorIfFieldsBefore(" / ")
            .appendHours().appendSuffix(":")
            .appendMinutes().appendSuffix("'")
            .appendSeconds().appendSuffix("\"")
            .toFormatter()



    override fun format(input: ReadableDuration): String {
        return pf.print(input.toPeriod())
    }
}